File: unity/Assets/Scripts/Multiplayer/Rooms/RoomJoinResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Rooms
{
    /// <summary>
    /// Response returned when a player successfully joins a game room.
    /// </summary>
    [System.Serializable]
    public class RoomJoinResponse : ApiResponse<ERoomJoinError>
    {
        /// <summary>
        /// The ID of the game room that was joined.
        /// </summary>
        public string room_id;

        /// <summary>
        /// Confirmation message for the join operation.
        /// </summary>
        public string message;
    }
}

File: java/src/main/java/com/michitai/multiplayer/rooms/RoomJoinResponse.java
package com.michitai.multiplayer.rooms;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

/**
 * Response returned when a player successfully joins a game room.
 */
public class RoomJoinResponse extends ApiResponse {
    @JsonProperty("room_id")
    private String roomId;

    @JsonProperty("message")
    private String message;

    public String getRoomId() {
        return roomId;
    }

    public void setRoomId(String roomId) {
        this.roomId = roomId;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}
